src/controllers/objecting.entity.name.controller.ts

```diff
@@ -1,27 +1,66 @@
 import { NextFunction, Request, Response } from "express";
+import { check, ValidationError, validationResult } from "express-validator";
+import { ErrorMessages } from "../model/error.messages";
+import { createGovUkErrorData, GovUkErrorData } from "../model/govuk.error.data";
 import { OBJECTIONS_COMPANY_NUMBER } from "../model/page.urls";
 import { addObjectionCreateToObjectionSession } from "../services/objection.session.service";
 import { ObjectionCreate } from "../modules/sdk/objections";
 import { Session } from "ch-node-session-handler";
+import { Templates } from "../model/template.paths";
+
+const FULL_NAME_FIELD: string = "fullName";
+const DIVULGE_INFO_FIELD: string = "shareIdentity";
+
+const validators = [
+  check(FULL_NAME_FIELD).not().isEmpty().withMessage(ErrorMessages.ENTER_FULL_NAME),
+  check(DIVULGE_INFO_FIELD).not().isEmpty().withMessage(ErrorMessages.SELECT_TO_DIVULGE),
+];
 
 /**
- * POST controller for objecting entity name screen
+ * ProcessForm validates input and processes form
  * @param req
  * @param res
  * @param next
  */
-export const post = (req: Request, res: Response, next: NextFunction) => {
-  const shareIdentityString: string = req.body.shareIdentity;
-  const fullNameValue: string = req.body.fullName;
-  const shareIdentityValue: boolean = shareIdentityString === "yes";
-  const objectionCreate: ObjectionCreate = { shareIdentity: shareIdentityValue, fullName: fullNameValue };
+export const processForm = [...validators, async (req: Request, res: Response, next: NextFunction): Promise<void> => {
```

Comment:
function needs to use an await if you declare it as async, therefore you can change function signature to 
export const processForm = [...validators, (req: Request, res: Response, next: NextFunction) => {

to remove the async and Promise<void> as we are not using await